code/src/components/Form.js

```diff
@@ -0,0 +1,106 @@
+import SubmittedView from "components/SubmittedView";
+import "./Form.css";
+
+const questions = [
+  {
+    id: "preference",
+    question_text: "What is your dietary preference",
+    input_type: "select",
+    options: [
+      { value: "Eat everything", label: "Eat everything" },
+      { value: "Pescatarian", label: "Pescatarian" },
+      { value: "Vegetarian", label: "Vegetarian" },
+      { value: "Vegan", label: "Vegan" },
+    ],
+  },
+  {
+    id: "cuisine",
+    question_text: "Which of the following cuisines are you craving most?",
+    input_type: "radio",
+    options: [
+      { value: "Persian cuisine", label: "Persian" },
+      { value: "Mexican cuisine", label: "Mexican" },
+      { value: "Japanese cuisine", label: "Japanese" },
+      { value: "Any cuisine", label: "Anything goes" },
+    ],
+  },
+  {
+    id: "spiciness",
+    question_text: "How well can you handle spice?",
+    input_type: "range",
+    format_answer: (value) => `Spice level: ${value} ${Number(value) === 1 ? "chili" : "chilis"}`,
+    options: [
+      { value: 1, label: "\u{1F336}" },
+      { value: 2, label: "\u{1F336}\u{1F336}" },
+      { value: 3, label: "\u{1F336}\u{1F336}\u{1F336}" },
+    ],
+  },
+  {
+    id: "herbs",
+    question_text: "Which of these herbs is superior?",
+    input_type: "select",
+    format_answer: (value) => `Heaps of ${value}`,
+    options: [
+      { value: "coriander", label: "Coriander" },
+      { value: "basil", label: "Basil" },
+      { value: "mint", label: "Mint" },
+      { value: "dill", label: "Dill" },
+      { value: "parsley", label: "Parsley" },
+    ],
+  },
+  {
+    id: "allergies",
+    question_text: "Do you have an allergies we should know about?",
+    input_type: "text",
+    format_answer: (value) => `Allergies: ${value.charAt(0).toUpperCase() + value.slice(1)}`,
+  },
+];
+
+const Form = () => {
+  const [currentQuestionIndex, setQuestionIndex] = useState(0);
+  const [answers, setAnswers] = useState([]);
+  const [inputValue, setInputValue] = useState("");
+
+  const buttonLabel = currentQuestionIndex === questions.length - 1 ? "Submit" : "Next question";
+
+  const handleInputChange = (event) => setInputValue(event.target.value);
+
+  const formatInput = (value) =>
+    questions[currentQuestionIndex].format_answer
+      ? questions[currentQuestionIndex].format_answer(value)
+      : value;
+
+  const handleOnSubmit = (event) => {
+    event.preventDefault();
+    setAnswers((arr) => [...arr, formatInput(inputValue)]);
+    setQuestionIndex(currentQuestionIndex + 1);
+    setInputValue("");
+  };
+
+  const didSubmit = () => questions.length === answers.length;
+
+  const resetForm = () => {
+    setAnswers([]);
+    setQuestionIndex(0);
+  };
+
+  if (didSubmit()) {
+    return <SubmittedView answers={answers} reset={resetForm} />;
+  }
+
+  return (
+    <form onSubmit={handleOnSubmit} className="container">
+      <Question
+        {...questions[currentQuestionIndex]}
```

Comment:
This solution is interesting! It's short and cool, but it also a bit implicit. I'd probably try to just use regular prop inputs to have a more readable API from the consumer:
```
const currentQuestion = questions[currentQuestionIndex]

     <Question
        question={currentQuestion}
        inputValue={inputValue}
        onChange={handleInputChange}
      />

Reply:
Thanks for the suggestion! Cool was not the intention but a side effect of being introduced to spread syntax last week :) I can test using regular props inputs as well. 

Also I see it probably makes sense either way to have a const like `currentQuestion` in your example to improve readability in other places where I have `questions[currentQuestionIndex]`